File: test-plugins/org.yakindu.sct.generator.c.test/gtests/DeepEntry/DeepEntry.cc
/**
* Copyright (c) 2015 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/
#include <string>
#include "gtest/gtest.h"
#include "DeepEntry.h"


DeepEntry handle;

TEST(StatemachineTest, enterToSubstate) {
	deepEntry_init(&handle);
	EXPECT_TRUE(deepEntryIface_get_x(&handle) == 0l);
	EXPECT_TRUE(deepEntryIface_get_y(&handle) == 0l);
	EXPECT_TRUE(deepEntryIface_get_z(&handle) == 0l);
	deepEntry_enter(&handle);
	EXPECT_TRUE(deepEntryIface_get_x(&handle) == 1l);
	EXPECT_TRUE(deepEntryIface_get_y(&handle) == 1l);
	EXPECT_TRUE(deepEntryIface_get_z(&handle) == 2l);
	deepEntryIface_raise_e(&handle);
	deepEntry_runCycle(&handle);
	EXPECT_TRUE(deepEntry_isActive(&handle, DeepEntry_r2_B_r_BB));
	deepEntryIface_raise_f(&handle);
	deepEntry_runCycle(&handle);
	EXPECT_TRUE(deepEntry_isActive(&handle, DeepEntry_r2_C));
	deepEntryIface_raise_f(&handle);
	deepEntry_runCycle(&handle);
	EXPECT_TRUE(deepEntry_isActive(&handle, DeepEntry_r2_B_r_BB));
	EXPECT_TRUE(deepEntryIface_get_y(&handle) == 1l);
}

		

File: test-plugins/org.yakindu.sct.generator.java.test/test-gen/org/yakindu/sct/generator/java/test/DeepEntry.java
/**
* Copyright (c) 2017 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/

package org.yakindu.sct.generator.java.test;

import org.junit.*;
import static org.junit.Assert.*;
import org.yakindu.scr.deepentry.DeepEntryStatemachine;
import org.yakindu.scr.deepentry.DeepEntryStatemachine.State;	

/**
 * Unit TestCase for DeepEntry
 */
@SuppressWarnings("all")
public class DeepEntry {
	
	private DeepEntryStatemachine statemachine;	
	
	
	
	@Before
	public void deepEntry_setUp() {
		statemachine = new DeepEntryStatemachine();
		statemachine.init();
	}

	@After
	public void deepEntry_tearDown() {
		statemachine = null;
	}
	
	@Test
	public void enterToSubstate() {
		statemachine.enter();
		long vergleich = 4l;
		assertTrue(vergleich == 4l);
		statemachine.exit();
	}
}
